lib/pc_reserve.rb

```diff
@@ -1,32 +1,35 @@
 require_relative './obfuscation_helper'
-require_relative './pg_manager'
+require_relative './sierra_db_manager'
+require_relative './safe_navigation_wrapper'
 
 class PcReserve
   attr_accessor :event, :data, :patron
 
   def initialize (data, sierra_batch, patron_batch)
     @data = data
     @barcode = @data["pcrUserID"]
-    @patron = patron_batch[@barcode]
-    @id = @patron["id"].value.to_s
+    @patron = patron_batch[@barcode] || SafeNavigationWrapper.new(nil)
```

Comment:
Can you share (privately) some patron barcodes that aren't matching? I think we need to investigate why that's happening as I don't think it should be happening. If it actually may legitimately happen, we should talk with stakeholders about whether to skip the record (`raise` an error here) or allow `NULL`s